In [3]:
# Week 5 - Anthony Hugan


#Your task in this week’s assignment is to answer three questions:

#1. What is the northernmost airport in the United States?

#2. What is the easternmost airport in the United States?

#3. On February 12th, 2013, which New York area airport had the windiest weather?


# this imports numpy and pandas and DataFrame / Series

import numpy as np

import pandas as pd

from pandas import DataFrame, Series

# sets pandas options

pd.set_option ('display.notebook_repr_html', True)

pd.set_option ('display.max_columns', None)

pd.set_option ('display.max_rows', None)

# Adding items for matplotlib

%matplotlib inline

import matplotlib.pyplot as plt

import matplotlib
matplotlib.style.use('ggplot')


In [29]:
# first we need to import the airports CSV file 

airports = pd.read_csv('airports.csv')

airports.head()

   faa                           name        lat        lon   alt  tz dst  \
0  04G              Lansdowne Airport  41.130472 -80.619583  1044  -5   A   
1  06A  Moton Field Municipal Airport  32.460572 -85.680028   264  -6   A   
2  06C            Schaumburg Regional  41.989341 -88.101243   801  -6   A   
3  06N                Randall Airport  41.431912 -74.391561   523  -5   A   
4  09J          Jekyll Island Airport  31.074472 -81.427778    11  -5   A   

              tzone  
0  America/New_York  
1   America/Chicago  
2   America/Chicago  
3  America/New_York  
4  America/New_York  

In [50]:
#1. What is the northernmost airport in the United States?

airports.sort_values('lat', ascending=False).head()


     faa                                        name        lat         lon  \
417  EEN                     Dillant Hopkins Airport  72.270833   42.898333   
230  BRW                  Wiley Post Will Rogers Mem  71.285446 -156.766003   
110  AIN                          Wainwright Airport  70.638056 -159.994722   
708  K03                               Wainwright As  70.613378 -159.860350   
152  ATK  Atqasuk Edward Burnell Sr Memorial Airport  70.467300 -157.436000   

     alt  tz dst              tzone  
417  149  -5   A                NaN  
230   44  -9   A  America/Anchorage  
110   41  -9   A  America/Anchorage  
708   35  -9   A  America/Anchorage  
152   96  -9   A  America/Anchorage  

When sorting by latitude in the CSV file it appears that Dillant Hopkins Airport is the northern most airport. After looking this up it appears the information in the CSV file is incorrect.  According to google maps the latitude and longitude for this airport is 42.906516347203976, -72.27290230930154, which means Wiley Post-Will Rogers Memorial Airport would be the northern most airport in this file. 

In [32]:
#2. What is the easternmost airport in the United States?

airports.sort_values('lon', ascending=False).head()

      faa                           name        lat         lon   alt  tz dst  \
1290  SYA                   Eareckson As  52.712275  174.113620    98  -9   A   
942   MYF               Montgomery Field  32.475900  117.759000    17   8   A   
396   DVT  Deer Valley Municipal Airport  33.411700  112.457000  1478   8   A   
417   EEN        Dillant Hopkins Airport  72.270833   42.898333   149  -5   A   
444   EPM     Eastport Municipal Airport  44.910111  -67.012694    45  -5   A   

                  tzone  
1290  America/Anchorage  
942      Asia/Chongqing  
396      Asia/Chongqing  
417                 NaN  
444    America/New_York  

To figure out the easternmost airport I sorted by longitude instead of latitude. According to this data Eareckson As is the eastern most airport. After looking this up in google maps it appears to be the correct latitude and longitude at 52.71466680688258, 174.09687229971834

In [38]:
# first we need to import the weather CSV file 

weather = pd.read_csv('weather.csv')

weather.head()

  origin  year  month  day  hour   temp   dewp  humid  wind_dir  wind_speed  \
0    EWR  2013      1    1     1  39.02  26.06  59.37     270.0    10.35702   
1    EWR  2013      1    1     2  39.02  26.96  61.63     250.0     8.05546   
2    EWR  2013      1    1     3  39.02  28.04  64.43     240.0    11.50780   
3    EWR  2013      1    1     4  39.92  28.04  62.21     250.0    12.65858   
4    EWR  2013      1    1     5  39.02  28.04  64.43     260.0    12.65858   

   wind_gust  precip  pressure  visib             time_hour  
0        NaN     0.0    1012.0   10.0  2013-01-01T06:00:00Z  
1        NaN     0.0    1012.3   10.0  2013-01-01T07:00:00Z  
2        NaN     0.0    1012.5   10.0  2013-01-01T08:00:00Z  
3        NaN     0.0    1012.2   10.0  2013-01-01T09:00:00Z  
4        NaN     0.0    1011.9   10.0  2013-01-01T10:00:00Z  

In [49]:
#3. On February 12th, 2013, which New York area airport had the windiest weather?


weather[(weather.wind_speed >= 20) & (weather.month == 2 ) & (weather.day == 12 ) & (weather.year == 2013)].sort_values(by=["wind_speed"], ascending=False).head()

      origin  year  month  day  hour   temp   dewp  humid  wind_dir  \
1009     EWR  2013      2   12     3  39.02  26.96  61.63     260.0   
18417    LGA  2013      2   12     2  42.98  26.06  50.94     290.0   
1018     EWR  2013      2   12    12  44.06  26.06  48.87     270.0   
18428    LGA  2013      2   12    13  44.06  23.00  43.02     300.0   
1008     EWR  2013      2   12     2  39.92  28.04  62.21     270.0   

       wind_speed  wind_gust  precip  pressure  visib             time_hour  
1009   1048.36058        NaN     0.0    1008.3   10.0  2013-02-12T08:00:00Z  
18417    23.01560   31.07106     0.0    1007.1   10.0  2013-02-12T07:00:00Z  
1018     21.86482   31.07106     0.0    1012.5   10.0  2013-02-12T17:00:00Z  
18428    21.86482   25.31716     0.0    1011.7   10.0  2013-02-12T18:00:00Z  
1008     20.71404   25.31716     0.0    1007.8   10.0  2013-02-12T07:00:00Z  

Next, to find wind speed on a specific date I filtered windspeed greater than 20, and the days February 12th 2013. This tells me EWR had a high windspeed of 1048 mph hour, which seems really high, considering the highest wind speed recorded was 231 MPH. The second highest in this CSV must be the correct answer at LGA, of 23 miles per hour. 
